src/Interpreters/DatabaseCatalog.cpp

```diff
@@ -522,9 +637,26 @@ void DatabaseCatalog::updateDatabaseName(const String & old_name, const String &
 
 DatabasePtr DatabaseCatalog::getDatabase(const String & database_name) const
 {
-    std::lock_guard lock{databases_mutex};
-    assertDatabaseExistsUnlocked(database_name);
-    return databases.find(database_name)->second;
+    DatabasePtr db;
+    {
+        std::lock_guard lock{databases_mutex};
+        db = databases.find(database_name)->second;
```

Comment:
this will not work - what if `database_name` doesn't exist in `databases`? it will dereference 'end' iterator...